core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -0,0 +1,286 @@
+   * list of its acls as a json object. Following gives an example:
+   *
+   * <pre>
+   * /kafka-acl/Topic/topic-1 => {"version": 1, "acls": [ { "host":"host1", "permissionType": "Allow","operation": "Read","principal": "User:alice"}]}
+   * /kafka-acl/Cluster/kafka-cluster => {"version": 1, "acls": [ { "host":"host1", "permissionType": "Allow","operation": "Read","principal": "User:alice"}]}
+   * /kafka-acl/ConsumerGroup/group-1 => {"version": 1, "acls": [ { "host":"host1", "permissionType": "Allow","operation": "Read","principal": "User:alice"}]}
+   * </pre>
+   */
+  val AclZkPath = "/kafka-acl"
+
+  //notification node which gets updated with the resource name when acl on a resource is changed.
+  val AclChangedZkPath = "/kafka-acl-changed"
+
+  val AclChangedPrefix = "aclChanged"
+}
+
+class SimpleAclAuthorizer extends Authorizer with Logging {
+  private var superUsers = Set.empty[KafkaPrincipal]
+  private var shouldAllowEveryoneIfNoAclIsFound: Boolean = _
+  private var zkClient: ZkClient = _
+  private var aclChangeListener: ZkNodeChangeNotificationListener = _
+
+  private val aclCache: scala.collection.mutable.HashMap[Resource, Set[Acl]] = new scala.collection.mutable.HashMap[Resource, Set[Acl]]
+  private val lock = new ReentrantReadWriteLock()
+
+  /**
+   * Guaranteed to be called before any authorize call is made.
+   */
+  override def configure(configs: util.Map[String, _]): Unit = {
+    val config = configs.asScala
+    val props = new java.util.Properties()
+    config foreach { case (key, value) => props.put(key, value.toString) }
+    val kafkaConfig = KafkaConfig.fromProps(props)
+
+    superUsers = configs.get(SimpleAclAuthorizer.SuperUsersProp) match {
+      case null => Set.empty[KafkaPrincipal]
+      case (str: String) => if (!str.isEmpty) str.split(",").map(s => KafkaPrincipal.fromString(s.trim)).toSet else Set.empty
+      case _ => Set.empty
+    }
+
+    shouldAllowEveryoneIfNoAclIsFound = if (config.contains(SimpleAclAuthorizer.AllowEveryoneIfNoAclIsFoundProp))
+      configs.get(SimpleAclAuthorizer.AllowEveryoneIfNoAclIsFoundProp).toString.toBoolean
+    else false
+
+    val zkUrl = config.getOrElse(SimpleAclAuthorizer.ZkUrlProp, kafkaConfig.zkConnect).toString
+
+    zkClient = ZkUtils.createZkClient(zkUrl, Integer2int(kafkaConfig.zkConnectionTimeoutMs), Integer2int(kafkaConfig.zkConnectionTimeoutMs))
+    ZkUtils.makeSurePersistentPathExists(zkClient, SimpleAclAuthorizer.AclZkPath)
+
+    aclChangeListener = new ZkNodeChangeNotificationListener(zkClient, SimpleAclAuthorizer.AclChangedZkPath, SimpleAclAuthorizer.AclChangedPrefix, AclChangedNotificaitonHandler)
+    aclChangeListener.startup()
+
+    zkClient.subscribeStateChanges(ZkStateChangeListener)
+
+    loadCache
+  }
+
+  override def authorize(session: Session, operation: Operation, resource: Resource): Boolean = {
+    val principal: KafkaPrincipal = new KafkaPrincipal(KafkaPrincipal.USER_TYPE, session.principal.getName)
```

Comment:
I had a comment in the jira. It seems that in SocketServer, when creating the session object, it's better to create a KafkaPrincipal instead of using KafkaChannel.principal(). The type in KafkaPrincipal should always be USER and the name should be KafkaChannel.principal().getName(). Then, we can just get the KafkaPrincipal from Session and don't need to create a new one.